### PR TITLE
NN-4858: source Renovate config from shared HMPPS repo

### PR DESCRIPTION
**Pull request by neilmendum**
https://dsdmoj.atlassian.net/browse/NN-4858

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -1,37 +1,37 @@
 {
-  "extends": ["config:base", ":disableDependencyDashboard", ":preserveSemverRanges"],
-  "assigneesFromCodeOwners": true,
-  "timezone": "Europe/London",
-  "schedule": ["after 9am every weekday", "before 5pm every weekday"],
-  "enabledManagers": ["npm", "nvm", "dockerfile", "docker-compose", "helmv3", "helm-values", "circleci"],
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "extends": ["github>ministryofjustice/hmpps-renovate-config:node"],
+  "reviewers": ["neilmendum", "timoj58", "natclamp-moj"],
```
Comment (neilmendum) [inline renovate.json:4]: As we don't use `CODEOWNERS` so configuring Renovate to assign PRs to each of the team individually

File: renovate.json [hunk 2]
```diff
@@ -1,37 +1,37 @@
...
     {
       "matchManagers": ["npm"],
-      "matchPackageNames": ["@types/node"],
-      "matchUpdateTypes": ["major"],
-      "enabled": false
-    },
-    {
-      "matchDatasources": ["docker"],
-      "allowedVersions": "16.15-bullseye-slim"
+      "rangeStrategy": "bump"
```
Comment (neilmendum) [inline renovate.json:8]: As per [Renovate docs](https://docs.renovatebot.com/configuration-options/#rangestrategy)

bump = e.g. bump the range even if the new version satisfies the existing range, e.g. ^1.0.0 -> ^1.1.0

This should avoid us needing to manually bump minor/patch dependencies manually

File: renovate.json [hunk 3]
```diff
@@ -1,37 +1,37 @@
...
     },
     {
-      "matchDatasources": ["docker-compose"],
-      "matchPackageNames": ["bitnami/redis"],
-      "allowedVersions": "5.0"
+      "matchManagers": ["npm"],
+      "matchUpdateTypes": ["minor", "patch"],
+      "groupName": "all non major NPM dependencies",
+      "groupSlug": "all-npm-minor-patch",
+      "stabilityDays": 3
```
Comment (neilmendum) [inline renovate.json:15]: Have introduced `stabilityDays`

> Number of days required before a new release is considered stable.

and `groupName` - to group them together in a single PR once they are at least 3 days old - this should ensure we aren't continually swamped with PRs. I spent some time testing this config on https://github.com/ministryofjustice/renovate-typescript-poc/pulls but can see how we find it?

File: renovate.json [hunk 4]
```diff
@@ -1,37 +1,37 @@
...
-      "matchPackageNames": ["govuk-frontend"],
-      "rangeStrategy": "bump"
+      "matchDatasources": ["docker-compose"],
+      "matchPackageNames": ["bitnami/redis"],
+      "allowedVersions": "5.0"
     }
-  ]
+  ],
+  "vulnerabilityAlerts": {
+    "stabilityDays": 0
```
Comment (neilmendum) [inline renovate.json:35]: As per [Renovate docs](https://docs.renovatebot.com/configuration-options/#vulnerabilityalerts)

> Renovate can read GitHub's Vulnerability Alerts to customize its Pull Requests

So configure Renovate to raise PRs for these straight away

File: renovate.json [hunk 5]
```diff
@@ -1,37 +1,37 @@
...
+      "groupSlug": "all-npm-minor-patch",
+      "stabilityDays": 3
     },
     {
-      "matchDatasources": ["orb"],
-      "matchPackageNames": ["cimg/node"],
-      "allowedVersions": "16.15-browsers"
+      "matchPackageNames": ["typescript", "govuk-frontend"],
+      "rangeStrategy": "bump",
+      "stabilityDays": 0
```
Comment (neilmendum) [inline renovate.json:20]: Set this to 0 so we bump these straight away, as we have the `security` pipeline job `npm_outdated`

File: renovate.json [hunk 6]
```diff
@@ -1,37 +1,37 @@
 {
-  "extends": ["config:base", ":disableDependencyDashboard", ":preserveSemverRanges"],
-  "assigneesFromCodeOwners": true,
-  "timezone": "Europe/London",
-  "schedule": ["after 9am every weekday", "before 5pm every weekday"],
-  "enabledManagers": ["npm", "nvm", "dockerfile", "docker-compose", "helmv3", "helm-values", "circleci"],
```
Comment (neilmendum) [inline renovate.json:6]: We now inherit this from https://github.com/ministryofjustice/hmpps-renovate-config/blob/4acc8951af908562c507a76b68ead835193c6a31/node.json#L4

**Review by natclamp-moj — APPROVED** [review]

**Review by natclamp-moj — APPROVED** [review]

**Pull request merged**